### PR TITLE
Feature/nix from buildpackages

### PR DESCRIPTION
**Pull request by angerman**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by angerman** [comment]
@Ericson2314 what do you think? Plausible?

**Comment by Ericson2314** [comment]
Err these are runtime dependencies of cabal2nix, right? If so, I don't think this is the right approach. My guess is that there is a `patchShebangs` messing up the prefetch script build, and I think @dtzWill recently fixed something with the Nix cross compile but maybe did so on staging.

**Comment by angerman** [comment]
When would I want to use the them *not* from the buildPackages? After all nix is run on the build machine, no?

**Comment by Ericson2314** [comment]
>  After all nix is run on the build machine, no?

Ahhh. Yeah never think about where the current package *will* be used, because assumptions on the downstream will cripple composition. Instead, since cabal2nix depends on nix at run time, cabal2nix and cabal2nix's nix must come from the same bootstrapping stage. If we are doing a `callCabal2nix` on the current machine, yes `cabal2nix` and `nix` would come from `buildPackages` and run on the build machine. But if we are cross-compiling an entirely NixOS with `cabal2nix` pre-installed globally, both come from `pkgs` as both will run on the host machine.

In short, `buildPackages.cabal2nix -> buildPackages.nix`, `pkgs.cabal2nix -> pkgs.nix`, `targetPackages.cabal2nix -> pkgs.nix`.
